tubular/nominal.py

```diff
@@ -445,19 +445,31 @@ def transform(self, X: pd.DataFrame) -> pd.DataFrame:
             # for categorical dtypes have to set new category for the impute values first
             # and convert back to the categorical type, other it will convert to object
             if "category" in X[c].dtype.name:
+
+                categories_before = X[c].dtype.categories
+
                 if self.rare_level_name not in X[c].cat.categories:
                     X[c] = X[c].cat.add_categories(self.rare_level_name)
 
-                dtype_before = X[c].dtype
-
                 X[c] = pd.Series(
                     data=np.where(
                         X[c].isin(self.non_rare_levels[c]),
                         X[c],
                         self.rare_level_name,
                     ),
                     index=X.index,
-                ).astype(dtype_before)
+                )
+
+                X[c] = pd.Categorical(
+                    X[c],
+                    categories=categories_before.tolist() + [self.rare_level_name],
+                )
+                rare_categories = [
```

Comment:
could you remove the rare_categories from categories_before (making e.g. output_categories) before creating the categorical on line 463?

Feels like it makes more sense to create the column from the minimal list rather than create first and remove later

Reply:
good suggestion, implemented here https://github.com/lvgig/tubular/pull/177/commits/66034aebe3167d2521b7988d5c5de08193987f12